CONTRIBUTING.md

```diff
@@ -2,10 +2,269 @@
+type ExampleCreateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,tasks"`
+
+	// Required: The name of the example
+	Name string `jsonapi:"attr,name"`
+
+	// Required: The URL to send in the example
+	URL string `jsonapi:"attr,url"`
+
+	// Optional: An optional value that is omitted if empty
+	OptionalValue *string `jsonapi:"attr,optional-value,omitempty"`
+}
+
+func (o *ExampleCreateOptions) valid() error {
+	if !validString(&o.Name) {
+		return ErrRequiredName
+	}
+
+	if !validString(&o.URL) {
+		return ErrInvalidRunTaskURL
+	}
+
+	return nil
+}
+
+// Create is used to create a new example for an organization
+func (s *example) Create(ctx context.Context, organization string, options ExampleCreateOptions) (*Example, error) {
+	if !validStringID(&organization) {
+		return nil, ErrInvalidOrg
+	}
+
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("organizations/%s/tasks", url.QueryEscape(organization))
+	req, err := s.client.newRequest("POST", u, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	r := &Example{}
+	err = s.client.do(ctx, req, r)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// A list of relations to include with an example. See available resources:
+// https://www.terraform.io/cloud-docs/api-docs/examples#list-examples (replace this URL with the actual documentation URL)
+type ExampleIncludeOps string
+
+const (
+	ExampleOrganization ExampleIncludeOps = "organization"
+)
+
+// ExampleListOptions represents the set of options for listing examples
+type ExampleListOptions struct {
+	ListOptions
+
+	// A list of relations to include
+	Include []ExampleIncludeOps `url:"include,omitempty"`
+}
+
+// List all the examples for an organization
+func (s *example) List(ctx context.Context, organization string, options *ExampleListOptions) (*ExampleList, error) {
+	if !validStringID(&organization) {
+		return nil, ErrInvalidOrg
+	}
+
+	u := fmt.Sprintf("organizations/%s/examples", url.QueryEscape(organization))
+	req, err := s.client.newRequest("GET", u, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	el := &ExampleList{}
+	err = s.client.do(ctx, req, el)
+	if err != nil {
+		return nil, err
+	}
+
+	return el, nil
+}
+
+// Read is used to read an organization's example by ID
+func (s *example) Read(ctx context.Context, exampleID string) (*Example, error) {
+	return s.ReadWithOptions(ctx, exampleID, nil)
+}
+
+// ExampleReadOptions represents the set of options for reading an example
+type ExampleReadOptions struct {
+	Include []RunTaskIncludeOps `url:"include,omitempty"`
+}
+
+// Read is used to read an organization's example by ID with options
+func (s *example) ReadWithOptions(ctx context.Context, exampleID string, options *ExampleReadOptions) (*Example, error) {
```

Comment:
Maybe we can make a reference here of the new rule we were discussing with Sebastian on our proj-go-tfe-1-0 channel: If the ID required is going to be composed of more than two individual strings, say `ReadWithOptions(ctx context.Context, exampleID, example2ID, example3ID string, options *ExampleReadOptions)`, then we prefer this to be passed as a single struct that contains the ID parts as fields for that struct.  like this example: https://github.com/hashicorp/go-tfe/blob/546d40891d90d05d42550682c8735d7dc9428bca/registry_provider.go#L173-L178

Reply:
Does it need to be decorated as jsonapi in your example, or wouldn't we expect a plain ol struct?

Reply:
I don't really want to complicate the example to accommodate multiple IDs